lib/size-chunker.js

```diff
@@ -89,19 +91,20 @@
 
          if (this._bytesPassed + chunk.length < this._chunkSize) {
 
-            this._startIfNeededAndPushData(chunk);
-            done();
+            this._startIfNeededAndPushData(chunk, done);
          } else {
 
             remainder = bytesLeave - chunk.length;
 
             if (remainder === 0) {
-               this._startIfNeededAndPushData(chunk);
-               this._finishChunk(done);
+               this._startIfNeededAndPushData(chunk, function() {
+                  this._finishChunk(done);
+               }.bind(this));
```

Comment:
May be we need to swap order here? Now we can get `finishChunk` event for chunk N after `startChunk` for N+1